test/lib/redux-cozy-client/adapters/CozyStackAdapter.spec.js

```diff
@@ -0,0 +1,41 @@
+/* eslint-env jest */
+import CozyStackAdapter from '../../../../src/lib/redux-cozy-client/adapters/CozyStackAdapter'
+
+describe('CozyStack Adapter', () => {
+  describe('fetchKonnectors', () => {
+    it('should ignore manifest id', async () => {
+      const stackClient = {
+        fetchJSON: async () => {
+          return {
+            data: [
+              {
+                slug: 'test',
+                id: 'io.cozy.konnectors/test',
+                attributes: {
+                  id: 'manifest id'
+                }
+              }
+            ]
+          }
+        }
+      }
+      const adapter = new CozyStackAdapter(stackClient)
+      const konnectors = await adapter.fetchKonnectors()
+      expect(konnectors).toEqual({
+        data: [
+          {
+            _type: 'io.cozy.konnectors',
+            attributes: {
+              id: 'manifest id'
+            },
+            id: 'io.cozy.konnectors/test',
+            slug: 'test'
+          }
+        ],
+        meta: undefined,
+        next: false,
+        skip: 0
+      })
+    })
+  })
+})
```

Comment:
Is it something that exist in cozy-client and we should backport? https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/KonnectorCollection.js 

I don't want to have this regression when dropping CozyStackAdapter

Reply:
I don't know. I suppose when we drop CozyStackAdapter we will also drop redux-cozy-client which is the one which fails because of this id attribute.

Reply:
KonnectorCollection extends AppCollection https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/KonnectorCollection.js#L12 

get from AppCollection is here: https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/AppCollection.js#L26

it seems we normalize the doc : https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/AppCollection.js#L72

And it seems that we merge attributes together https://github.com/cozy/cozy-client/blob/master/packages/cozy-stack-client/src/AppCollection.js#L13-L15 

And this app.attributes comes last, I think we override the real id of the document by the id present in the manifest itself. So I think we should fix the issue on cozy-client too since if we make a request client.get(Q('io.cozy.konnector').getById('io.cozy.konnector/test')) we will receive id:app manifest and not io.cozy.konnector.test. 

We have opened this door, let's fix it everywhere...

Reply:
https://github.com/cozy/cozy-client/pull/1026